pkgs/leak_tracker/test/tests/leak_tracking/_object_tracker_test.dart

```diff
@@ -375,96 +389,197 @@ void main() {
+    void checkContext(
+      Map<String, dynamic>? context, {
+      required bool shouldContain,
+      required String key,
+    }) {
+      if (shouldContain) {
+        expect(context!.keys, contains(key));
+      } else {
+        expect(context?.keys ?? [], isNot(contains(key)));
+      }
+    }
 
-    test('when object is disposed after phase ended.', () async {
-      const objectDisposedInPhase = '1';
-      const objectLeaking = '2';
-      const objectDisposedAfterPhase = '3';
-      final allObjects = [
-        objectDisposedInPhase,
-        objectLeaking,
-        objectDisposedAfterPhase,
-      ];
-
-      phase.value = const PhaseSettings(name: 'phase1');
-      allObjects.forEach(startTracking);
-      tracker.dispatchDisposal(objectDisposedInPhase, context: null);
-      phase.value = const PhaseSettings.paused();
-      tracker.dispatchDisposal(objectDisposedAfterPhase, context: null);
-
-      // GC all objects.
-      allObjects.forEach(finalizerBuilder.gc);
-
-      // Check leaks are collected only for leaking object that
-      // was registered in the phase.
-      final leaks = await tracker.collectLeaks();
-      expect(leaks.total, 1);
-      final theLeak = leaks.byType[LeakType.notDisposed]!.single;
-      expect(theLeak.code, identityHashCode(objectLeaking));
-    });
+    void checkNotDisposedLeak(
+      Leaks leaks,
+      PhaseSettings phase, {
+      required Object object,
+    }) {
+      var leak = leaks.notDisposed
+          .where((l) => l.code == identityHashCode(object))
+          .onlyOrNull;
+
+      if (phase.isPaused || phase.allowAllNotDisposed) {
+        expect(leak, isNull);
+        return;
+      }
+      leak = leak!;
+
+      expect(leak.phase, phase.name);
+
+      checkContext(
+        leak.context,
+        shouldContain: phase.leakDiagnosticConfig.collectStackTraceOnStart,
+        key: ContextKeys.startCallstack,
+      );
+    }
+
+    void checkNotGCedLeak(
+      Leaks leaks,
+      PhaseSettings phase, {
+      required Object object,
+    }) {
+      var leak = leaks.notGCed
+          .where((l) => l.code == identityHashCode(object))
+          .onlyOrNull;
+
+      if (phase.isPaused || phase.allowAllNotGCed) {
+        expect(leak, isNull);
+        return;
+      }
+      leak = leak!;
+
+      expect(leak.phase, phase);
+
+      checkContext(
+        leak.context,
+        shouldContain: phase.leakDiagnosticConfig.collectStackTraceOnStart,
+        key: ContextKeys.startCallstack,
+      );
+      checkContext(
+        leak.context,
+        shouldContain: phase.leakDiagnosticConfig.collectStackTraceOnDisposal,
+        key: ContextKeys.disposalCallstack,
+      );
+      checkContext(
+        leak.context,
+        shouldContain:
+            phase.leakDiagnosticConfig.collectRetainingPathForNotGCed,
+        key: ContextKeys.retainingPath,
+      );
+    }
+
+    for (var gced in [
+      true,
+      //false,
+    ]) {
+      for (var disposed in [
+        //true,
```

Comment:
```suggestion
      false,
    ]) {
      for (var disposed in [
        true,
```
Should these be uncommented?

Reply:
yes!